Src/FluentAssertions/Collections/CollectionAssertions.cs

```diff
@@ -639,6 +639,117 @@ public AndConstraint<TAssertions> BeEquivalentTo(IEnumerable expectation, string
+        /// Notice that actual behavior is determined by the global defaults managed by <see cref="AssertionOptions"/>.
+        /// </remarks>
+        /// <param name="config">
+        /// A reference to the <see cref="EquivalencyAssertionOptions{TSubject}"/> configuration object that can be used
+        /// to influence the way the object graphs are compared. You can also provide an alternative instance of the
+        /// <see cref="EquivalencyAssertionOptions{TSubject}"/> class. The global defaults are determined by the
+        /// <see cref="AssertionOptions"/> class.
+        /// </param>
+        /// <param name="because">
+        /// An optional formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the
+        /// assertion is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> NotContainEquivalentOf<TExpectation>(TExpectation unexpected, Func<EquivalencyAssertionOptions<TExpectation>,
+            EquivalencyAssertionOptions<TExpectation>> config, string because = "", params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(config, nameof(config));
+
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(Subject != null)
+                .FailWith("Expected {context:collection} not to contain equivalent of {0}{reason}, but collection is <null>.", unexpected);
+
+            EquivalencyAssertionOptions<TExpectation> options = config(AssertionOptions.CloneDefaults<TExpectation>());
+
+            var foundIndices = new List<int>();
+            using (var scope = new AssertionScope())
+            {
+                int index = 0;
+                foreach (object actualItem in Subject)
+                {
+                    var context = new EquivalencyValidationContext
+                    {
+                        Subject = actualItem,
+                        Expectation = unexpected,
+                        CompileTimeType = typeof(TExpectation),
+                        Because = because,
+                        BecauseArgs = becauseArgs,
+                        Tracer = options.TraceWriter,
+                    };
+
+                    var equivalencyValidator = new EquivalencyValidator(options);
+                    equivalencyValidator.AssertEquality(context);
+
+                    string[] failures = scope.Discard();
+
+                    if (!failures.Any())
+                    {
+                        foundIndices.Add(index);
+                    }
+
+                    index++;
+                }
+            }
+
+            if (foundIndices.Count > 0)
+            {
+                using (new AssertionScope())
```

Comment:
I think this could be written as:
```c#
if (foundIndices.Count > 0)
{
    Assertion assertion = Execute.Assertion
        .BecauseOf(because, becauseArgs)
        .WithExpectation("Expected {context:collection} {0} not to contain equivalent of {1}{reason}, ", Subject, unexpected)
        .AddReportable("configuration", options.ToString());

    if (foundIndices.Count == 1)
    {
        assertion
            .FailWith("but found one at index {0}.", foundIndices[0])
            .Then
            .ClearExpectation();
    }
...
}
```

Reply:
I tried similar approach earlier and it did not work unfortunately. 

Firstly I was surprised that `AddReportable` is a `void` method, however I've extracted return value from `.WithExpectation` and just called `AddReportable` on it (that's not an issue of course).

The problem I faced is that `AddReportable` does not work when called outside `new AssertionScope()`. I'm not familiar with the codebase much, but quick look brought me to conclusion that "With" stuff is only attached on `Dispose` which seems to be not called on regular non-lazy assertion scope.

Am I missing something? What do you think?

Reply:
I haven't used `AddReportable` much, so @dennisdoomen might be better to answer this than me.

Reply:
Ok thanks.

BTW I added a test with using `NotContainEquivalentOf`  inside `AssertionScope`: https://github.com/fluentassertions/fluentassertions/pull/1318/commits/05b5f662771cf15fd1817b26268b74e81fba76d6. I did it in a separate commit, because I don't know how this aligns with your testing approach. Looks like it works fine (if I understand "expected" correctly).

Reply:
`AddReportable` was indeed only used by `EquivalencyValidator` to initialize an `AssertionScope` within which the entire equivalency engine runs.